src/Engine/ProtoCore/Parser/AssociativeAST.cs

```diff
@@ -433,6 +433,13 @@ public override string ToString()
 
     public class TypedIdentifierNode : IdentifierNode
     {
+        public TypedIdentifierNode()
```

Comment:
I added the constructor below:
`TypedIdentifierNode(IdentifierNode rhs)` in order to clone a `TypedIdentifierNode` into another `TypedIdentifierNode` in `NodeUtils.Clone`. Prior to this fix, a `TypedIdentifierNode` was simply being cloned into an `IdentifierNode` because of which the type information was being lost in pre-compilation. Now that this is fixed, we can see the type related warnings as expected such as in the following:
`a : int = Point.ByCoordinates();`. This will now throw a warning related to type-mismatch. This was not happening earlier. Since I added this constructor, I needed to also add the default constructor as it was being called in other places.